Bindings/Python/examples/posthoc_StatesTrajectory_example.py

```diff
@@ -0,0 +1,81 @@
+#                                                                         #
+# Copyright (c) 2005-2018 Stanford University and the Authors             #
+# Author(s): Christopher Dembia                                           #
+#                                                                         #
+# Licensed under the Apache License, Version 2.0 (the "License");         #
+# you may not use this file except in compliance with the License.        #
+# You may obtain a copy of the License at                                 #
+# http://www.apache.org/licenses/LICENSE-2.0.                             #
+#                                                                         #
+# Unless required by applicable law or agreed to in writing, software     #
+# distributed under the License is distributed on an "AS IS" BASIS,       #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or         #
+# implied. See the License for the specific language governing            #
+# permissions and limitations under the License.                          #
+# ----------------------------------------------------------------------- #
+
+# This example shows how to use the StatesTrajectory to analyze a simulation
+# (computing joint reaction forces).
+
+import opensim as osim
+import math
+
+model = osim.Model()
+model.setName('model')
+
+# Create a pendulum model.
+# ------------------------
+# In the default pose, the system is a mass hanging 1 meter down from a hinge.
+body = osim.Body('body', 1.0, osim.Vec3(0), osim.Inertia(1))
+model.addComponent(body)
+
+joint = osim.PinJoint('joint',
+        model.getGround(), osim.Vec3(0), osim.Vec3(0), 
+        body, osim.Vec3(0, 1.0, 0), osim.Vec3(0))
+joint.updCoordinate().setName('q')
+model.addComponent(joint)
+
+# This reporter will collect states throughout the simulation at intervals of
+# 0.05 seconds.
+rep = osim.StatesTrajectoryReporter()
+rep.setName('reporter')
+rep.set_report_time_interval(0.05)
+model.addComponent(rep)
+
+# Simulate.
+# ---------
+state = model.initSystem()
+
+# The initial position is that the pendulum is rotated 45 degrees
+# counter-clockwise from its default pose.
+model.setStateVariableValue(state, 'joint/q/value', 0.25 * math.pi)
+
+manager = osim.Manager(model, state)
+manager.integrate(1.0)
+
+# Analyze the simulation.
+# -----------------------
+# Retrieve the StatesTrajectory from the reporter. Alternately, we could load a
+# StatesTrajectory from a STO file using
+# StatesTrajectory.createFromStatesStorage().
+statesTraj = rep.getStates()
+
+for itime in range(statesTraj.getSize()):
+    state = statesTraj[itime]
+    time = state.getTime()
+    q = model.getStateVariableValue(state, 'joint/q/value')
+
+    # Calculating joint reactions requires realizing to Acceleration.
+    model.realizeAcceleration(state)
+    # This returns a SpatialVec, which is two Vec3's ([0]: moment, [1]: force).
+    reaction = joint.calcReactionOnParentExpressedInGround(state)
```

Comment:
Might also be a nice opportunity to use the joint's `Output` for the reaction force here.

Reply:
My understanding is the outputs are to be used with reporters (or for wiring inputs and outputs between components, in general). If accessing the quantity directly, there's no need to use the output and I can use the underlying method directly. Also, the concrete outputs are harder to access via scripting.

My preference would be to add the following (commented-out code):

```python
# The reaction can also be accessed as an output:
# abstractOutput = joint.getOutput("reaction_on_parent")
# output = osim.OutputSpatialVec.safeDownCast(abstractOutput)
# outputValue = output.getValue(s)
```